### DISCUSSION TITLE
How can I filter based on `tests/name.rs` ?

### DISCUSSION THREAD

**Discussion by erikschul** [nextest-rs/nextest#1724] [Q&A] [answered]
I have a number of tests, but I'm only interested in running specific integration tests, e.g. `tests/some_long_name.rs` which match "long".

I've tried:

```
cargo nextest run -E 'binary(long)'
cargo nextest run -E 'package(long)'
cargo nextest run -E 'test(long)'
cargo nextest run -E 'deps(long)'
```
and nothing matches.

But without any filter it runs normally, e.g.
```
PASS [   0.004s] myproject::some_long_name tests::some_test_name
```

And similarly I want to match tests by module name, e.g. `/src/module_name/submodule_name/mod.rs` containing `mod tests { fn some_test } `.
```
PASS [   0.091s] myproject module_name::submodule_name::tests::some_test
```

**How can I match the integration test name, or the module name?**

**Comment by sunshowers** [comment, marked as answer]
`'binary(*long*)'` or `'binary(~long)'` should work.

The default matcher for `binary` is the glob matcher -- see https://nexte.st/docs/filtersets/reference/#default-matchers. So you can either use a glob pattern (`*long*`) or switch to the contains matcher with `~`.
